elyra/pipeline/pipeline_definition.py

```diff
@@ -440,6 +458,44 @@ def validate(self) -> list:
 
         return validation_issues
 
+    def coerce_kv_pair_properties(self):
+        """
+        Convert pipeline- and node-level list properties that have been identified
+        as sets of key-value pairs from a plain list type to the KeyValueList list
+        type. Properties are identified as key-value pair properties when they set
+        keyValueEntries to true in the pipeline JSON canvas properties definition.
+        """
+        all_pipeline_properties = PipelineDefinition.get_canvas_properties_from_template(
+            package_name="templates/pipeline", template_name="pipeline_properties_template.jinja2"
+        )
+        kv_pair_properties = PipelineDefinition.get_key_value_pair_properties(all_pipeline_properties)
+
+        pipeline_default_properties = self.primary_pipeline.get_property(pipeline_constants.PIPELINE_DEFAULTS, {})
+        for pipeline_default_prop, pipeline_default_value in pipeline_default_properties.items():
+            if pipeline_default_prop in kv_pair_properties:
+                # Coerce plain list to KeyValueList
+                self.set_pipeline_default_property(pipeline_default_prop, KeyValueList(pipeline_default_value))
+
+        generic_node_properties = PipelineDefinition.get_canvas_properties_from_template(
+            package_name="templates/components", template_name="generic_properties_template.jinja2"
+        )
+        for node in self.pipeline_nodes:
+            if not Operation.is_generic_operation(node.op):
+                continue
+
+            for property_name, value in generic_node_properties["current_parameters"].items():
+                if property_name.startswith("elyra_"):
+                    property_name = property_name.replace("elyra_", "")
```

Comment:
I understand stripping off the `elyra_` prefix in order to check if this property is a KV pairs property, but that's only because the [`elyra_` prefix is stripped when building the `kv_pair_properties` list](https://github.com/elyra-ai/elyra/pull/2679/files#diff-79810d1ae48b2ef6e0b3b2602ffdf1ca2cf2a452f3695c67168495fbe73e3eafR649-R650) anyway.  Since we don't appear to persist the non-prefixed values, can we forgo both of these "stripping" locations?

I'm not finding any place in the old code where these prefixes are removed so just wondering why we need to do that.

Reply:
Unfortunately I don't think we can. Here and in the function you linked, we're taking property names directly from the templates where the `elyra_` prefix is present. The property name that we get during submit/export (and that we're checking for matches with the key-value pair property names), doesn't have the prefix - the frontend removes it at some point. 

I wonder if we could forego the `elyra_` prefix altogether at some point and just indicate this in the canvas JSON data stanza in some other way.

Reply:
I see.  Ok, no worries.
> I wonder if we could forego the elyra_ prefix altogether at some point and just indicate this in the canvas JSON data stanza in some other way.

This sounds like something that might be a helpful update at some point.  
cc: @ajbozarth